environs/__init__.py

```diff
@@ -176,6 +177,25 @@ def _preprocess_json(value: str, **kwargs):
     return pyjson.loads(value)
 
 
+_EnumT = typing.TypeVar("_EnumT", bound=Enum)
+
+
+def _enum_parser(value, *, type: typing.Type[_EnumT], ignore_case: bool = False, **kwargs) -> _EnumT:
+    invalid_exc = ma.ValidationError(f"Not a valid '{type.__name__}' enum.")
+
+    if not ignore_case:
+        try:
+            return type[value]
+        except KeyError:
+            raise invalid_exc
```

Comment:
nitpick:

```suggestion
        except KeyError as error:
            raise invalid_exc from error
```

Reply:
Maybe you missed my previous comment because I accidently resolved the issue.

> No need to do from error in an except clause as explained in this SO answer https://stackoverflow.com/a/24752607/5437493

Reply:
Ah, I did miss your previous comment.

Indeed, `from error` isn't strictly necessary; it just improves the error message a tad bit for debugging.

Anyway, it's certainly not a blocker for this PR.